client/signup/steps/welcome/index.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import StepWrapper from 'signup/step-wrapper';
+import SignupActions from 'lib/signup/actions';
+
+class WelcomeStep extends Component {
+	state = {
+		form: null,
+		submitting: false
+	};
+
+	handleSubmit = ( event ) => {
+		event.preventDefault();
+
+		const {
+			goToNextStep,
+			stepName,
+			translate,
+		} = this.props;
+
+		const timestamp = ( new Date() ).getTime();
+		const noise = Math.random().toString( 36 ).substring( 2, 10 );
+		const site = `site${ timestamp }${ noise }`;
```

Comment:
yes - good call - i will extract all this to a generateUniqueSite function and use uuid instead